packages/firestore/src/local/local_store.ts

```diff
@@ -1057,152 +1048,137 @@ export function newLocalStore(
-                txn,
-                keys
-              ) as PersistencePromise<MaybeDocumentMap | null>;
-            } else {
-              return PersistencePromise.resolve<MaybeDocumentMap | null>(null);
-            }
-          });
+        return targetCacheImpl
+          .getTargetDataForTarget(txn, targetId)
+          .next(targetData => (targetData ? targetData.target : null));
       }
     );
   }
+}
 
-  removeCachedMutationBatchMetadata(batchId: BatchId): void {
-    this.mutationQueue.removeCachedMutationKeys(batchId);
-  }
-
-  setNetworkEnabled(networkEnabled: boolean): void {
-    this.persistence.setNetworkEnabled(networkEnabled);
-  }
-
-  getActiveClients(): Promise<ClientId[]> {
-    return this.persistence.getActiveClients();
-  }
-
-  getTarget(targetId: TargetId): Promise<Target | null> {
-    const cachedTargetData = this.targetDataByTarget.get(targetId);
-
-    if (cachedTargetData) {
-      return Promise.resolve(cachedTargetData.target);
-    } else {
-      return this.persistence.runTransaction(
-        'Get target data',
-        'readonly',
-        txn => {
-          return this.targetCache
-            .getTargetDataForTarget(txn, targetId)
-            .next(targetData => (targetData ? targetData.target : null));
-        }
-      );
-    }
-  }
-
-  getNewDocumentChanges(): Promise<MaybeDocumentMap> {
-    return this.persistence
-      .runTransaction('Get new document changes', 'readonly', txn =>
-        this.remoteDocuments.getNewDocumentChanges(
-          txn,
-          this.lastDocumentChangeReadTime
-        )
+/**
+ * Returns the set of documents that have been updated since the last call.
+ * If this is the first call, returns the set of changes since client
+ * initialization. Further invocations will return document changes since
+ * the point of rejection.
+ */
+// PORTING NOTE: Multi-Tab only.
+export function getNewDocumentChanges(
+  localStore: LocalStore
+): Promise<MaybeDocumentMap> {
+  const localStoreImpl = debugCast(localStore, LocalStoreImpl);
+  const remoteDocumentCacheImpl = debugCast(
+    localStoreImpl.remoteDocuments,
+    IndexedDbRemoteDocumentCache // We only support IndexedDb in multi-tab mode.
+  );
+  return localStoreImpl.persistence
+    .runTransaction('Get new document changes', 'readonly', txn =>
+      remoteDocumentCacheImpl.getNewDocumentChanges(
+        txn,
+        localStoreImpl.lastDocumentChangeReadTime
       )
-      .then(({ changedDocs, readTime }) => {
-        this.lastDocumentChangeReadTime = readTime;
-        return changedDocs;
-      });
-  }
+    )
+    .then(({ changedDocs, readTime }) => {
+      localStoreImpl.lastDocumentChangeReadTime = readTime;
+      return changedDocs;
+    });
+}
 
-  async synchronizeLastDocumentChangeReadTime(): Promise<void> {
-    this.lastDocumentChangeReadTime = await this.persistence.runTransaction(
+/**
+ * Reads the newest document change from persistence and forwards the internal
```

Comment:
Nit: can you clarify where/to what it's being forwarded?

Reply:
The "forward" does not mean "send something along" but rather "move forward". Rewritten.